sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/NetworkSecurityRuleImpl.java

```diff
@@ -162,6 +173,9 @@ public NetworkSecurityRuleImpl fromAddress(String cidr) {
         this.innerModel().withSourceAddressPrefix(cidr);
         this.innerModel().withSourceAddressPrefixes(null);
         this.innerModel().withSourceApplicationSecurityGroups(null);
+        if (this.sourceAsgs != null) {
+            this.sourceAsgs.clear();
+        }
```

Comment:
what the purpose of this? seems not a new var?

Reply:
No it's not. 

`this.innerModel().withSourceApplicationSecurityGroups(null);` is not enough. When we update the rule type from `ApplicationSecurityGroups` to others (ipAdress, service tag, etc), we'll still be sending ApplicationSecurityGroup configuration if we don't clear the `sourceAsgs` or `destinationAsgs`, as my experience with the test.